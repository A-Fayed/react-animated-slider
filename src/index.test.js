import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Slider, { DEFAULT_CLASSNAMES as classNames } from './';

jest.useFakeTimers();

test('mounts', () => {
	const slider = <Slider />;
	expect(slider).toBeDefined();
});

test('renders one child', () => {
	const slider = ReactTestRenderer.create(<Slider><div /></Slider>);
	expect(slider).toBeDefined();
	expect(slider.toJSON()).toMatchSnapshot();
});

test('renders two children', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /></Slider>);
	expect(slider).toBeDefined();
	expect(slider.toJSON()).toMatchSnapshot();
});

test('renders three children', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /><div /></Slider>);
	expect(slider).toBeDefined();
	expect(slider.toJSON()).toMatchSnapshot();
});

test('with one child can not navigate to next', () => {
	const slider = ReactTestRenderer.create(<Slider><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().next();
	jest.runAllTimers();
	expect(initialSnapshot).toEqual(slider.toJSON());
	expect(slider.toJSON()).toMatchSnapshot();
});

test('with one child can not navigate to previuous', () => {
	const slider = ReactTestRenderer.create(<Slider><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().previous();
	jest.runAllTimers();
	expect(initialSnapshot).toEqual(slider.toJSON());
	expect(slider.toJSON()).toMatchSnapshot();
});

test('with two children can navigate to next', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().next();
	jest.runAllTimers();
	expect(initialSnapshot).not.toEqual(slider.toJSON());
	expect(slider.toJSON()).toMatchSnapshot();
});

test('with two children can navigate to next only once', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /></Slider>);
	slider.getInstance().next();
	jest.runAllTimers();
	const snapshot = slider.toJSON();
	slider.getInstance().next();
	jest.runAllTimers();
	expect(snapshot).toEqual(slider.toJSON());
	expect(slider.toJSON()).toMatchSnapshot();
});

test('with two children can not navigate to previuous', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().previous();
	jest.runAllTimers();
	expect(initialSnapshot).toEqual(slider.toJSON());
	expect(slider.toJSON()).toMatchSnapshot();
});

test('animate next classNames', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().next();
	const animatedSnapshot = slider.toJSON();
	expect(animatedSnapshot).toMatchSnapshot();
	expect(animatedSnapshot).not.toEqual(initialSnapshot);
});

test('animate previous classNames', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().previous();
	const animatedSnapshot = slider.toJSON();
	expect(animatedSnapshot).toMatchSnapshot();
	expect(animatedSnapshot).not.toEqual(initialSnapshot);
});

test('does not allow navigation while animating', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /><div /></Slider>);
	slider.getInstance().next();
	const animatedSnapshot = slider.toJSON();
	slider.getInstance().next();
	expect(animatedSnapshot).toEqual(slider.toJSON());
});

test('does not allow navigation if disabled', () => {
	const slider = ReactTestRenderer.create(<Slider disabled><div /><div /><div /></Slider>);
	const initialSnapshot = slider.toJSON();
	slider.getInstance().next();
	expect(initialSnapshot).toEqual(slider.toJSON());
	jest.runAllTimers(); 
	expect(initialSnapshot).toEqual(slider.toJSON());
});

test('test autoslide disabled', () => {
	const slider = ReactTestRenderer.create(<Slider><div /><div /></Slider>);
	expect(setInterval).toHaveBeenCalledTimes(0);
});

test('test autoslide enabled', () => {
	const slider = ReactTestRenderer.create(<Slider autoslide="1"><div /><div /></Slider>);
	expect(setInterval).toHaveBeenCalledTimes(1);
});




