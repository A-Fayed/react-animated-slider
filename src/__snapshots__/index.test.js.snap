// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`animate next classNames 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide animateOut next"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
    <div
      className="slide animateIn next"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
    <div
      className="slide hidden"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
  </div>
</div>
`;

exports[`animate previous classNames 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide animateOut previous"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
    <div
      className="slide hidden"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
    <div
      className="slide animateIn previous"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
  </div>
</div>
`;

exports[`renders one child 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide current"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
  </div>
</div>
`;

exports[`renders three children 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={false}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={false}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide current"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slide next"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slide previous"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;

exports[`renders two children 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={false}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide current"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slide next"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;

exports[`with one child can not navigate to next 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide current"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
  </div>
</div>
`;

exports[`with one child can not navigate to previuous 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide current"
      onTouchEnd={undefined}
      onTouchStart={undefined}
    />
  </div>
</div>
`;

exports[`with two children can navigate to next 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={false}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide previous"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slide current"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;

exports[`with two children can navigate to next only once 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={false}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={true}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide previous"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slide current"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;

exports[`with two children can not navigate to previuous 1`] = `
<div
  className={undefined}
>
  <button
    className="previousButton"
    disabled={true}
    onClick={[Function]}
  >
    previous
  </button>
  <button
    className="nextButton"
    disabled={false}
    onClick={[Function]}
  >
    next
  </button>
  <div
    className="track"
  >
    <div
      className="slide current"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slide next"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
  </div>
</div>
`;
